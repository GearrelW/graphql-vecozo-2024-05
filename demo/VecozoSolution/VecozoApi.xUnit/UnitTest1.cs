using HotChocolate.Execution;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;

namespace VecozoApi.xUnit
{
    public class UnitTest1
    {
        [Fact]
        public async Task Test1()
        {
            var schema = await new ServiceCollection()
                .AddGraphQL()
                .AddFiltering()
                .AddQueryType<QueryType>()
                .AddMutationType<MutationType>()
                .BuildSchemaAsync();

            schema.ToString().MatchSnapshot();
        }
    }
}