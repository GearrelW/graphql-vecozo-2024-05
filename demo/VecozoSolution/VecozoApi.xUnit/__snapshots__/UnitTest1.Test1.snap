schema {
  query: Query
  mutation: MutationType
}

type Episode {
  show: Show!
  id: Int!
  title: String!
  showId: Int!
}

type MutationType {
  addShow(input: AddShowInput): Show!
}

type Query {
  getShowById(id: Int!): Show!
  findShowByTitleAndReleaseYear(input: FindShowByTitleAndReleaseYearInput!): Show
  andertekstje: String!
  shows(where: ShowFilterInput): [Show!]!
  tekstje: String!
  favoriteShow: Show!
}

type Show {
  oogabooga: [Episode!]!
  id: Int!
  title: String!
  releaseDate: Date!
}

input AddShowInput {
  title: String!
  releaseDate: Date!
}

input DateOperationFilterInput {
  eq: Date
  neq: Date
  in: [Date]
  nin: [Date]
  gt: Date
  ngt: Date
  gte: Date
  ngte: Date
  lt: Date
  nlt: Date
  lte: Date
  nlte: Date
}

input EpisodeFilterInput {
  and: [EpisodeFilterInput!]
  or: [EpisodeFilterInput!]
  id: IntOperationFilterInput
  title: StringOperationFilterInput
  showId: IntOperationFilterInput
}

input FindShowByTitleAndReleaseYearInput {
  partOfTitle: String!
  releaseYear: Int!
}

input IntOperationFilterInput {
  eq: Int
  neq: Int
  in: [Int]
  nin: [Int]
  gt: Int
  ngt: Int
  gte: Int
  ngte: Int
  lt: Int
  nlt: Int
  lte: Int
  nlte: Int
}

input ListFilterInputTypeOfEpisodeFilterInput {
  all: EpisodeFilterInput
  none: EpisodeFilterInput
  some: EpisodeFilterInput
  any: Boolean
}

input ShowFilterInput {
  and: [ShowFilterInput!]
  or: [ShowFilterInput!]
  id: IntOperationFilterInput
  title: StringOperationFilterInput
  releaseDate: DateOperationFilterInput
  episodes: ListFilterInputTypeOfEpisodeFilterInput
}

input StringOperationFilterInput {
  and: [StringOperationFilterInput!]
  or: [StringOperationFilterInput!]
  eq: String
  neq: String
  contains: String
  ncontains: String
  in: [String]
  nin: [String]
  startsWith: String
  nstartsWith: String
  endsWith: String
  nendsWith: String
}

"The `Date` scalar represents an ISO-8601 compliant date type."
scalar Date
